File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numsSorted = nums.copy()
        numsSorted.sort()
        print(numsSorted)
        left = 0
        right = len(nums) - 1
        while (val := numsSorted[left] + numsSorted[right]) != target:
            if val > target:
                right-=1
            else:
                left+=1
        leftIndex = nums.index(numsSorted[left])
        rightIndex = 0
        if numsSorted[left] == numsSorted[right]:
            rightIndex = nums.index(numsSorted[right], leftIndex + 1)
        else:
            rightIndex = nums.index(numsSorted[right])
        
        if leftIndex < rightIndex:
            return [leftIndex, rightIndex]
        else:
            return [rightIndex, leftIndex]
        

        

File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {
        return recursiveSearch(nums, target, 0, nums.length - 1);
    }
    
    public int recursiveSearch(int[] nums, int target, int left, int right) {
        if(left > right) return -1;

        int midIndex = (right + left + 1) / 2;
        int val = nums[midIndex];
        if(val == target) {
            return midIndex;
        }
        else if(val > target) {
            return recursiveSearch(nums, target, left, midIndex - 1);
        } else {
            return recursiveSearch(nums, target, midIndex + 1, right);
        }
    }
}
